### ISSUE TITLE
Add icons to binaries built using build_apps

### ISSUE THREAD

**Issue by Moguri** [panda3d/panda3d#468]
Binaries built with the build_apps Setuptools command should have the option to have an icon specified. This is a feature that our Setuptools-based approach is missing from pdeploy. I believe we'll need to do the following for each platform:

- [x] Windows: Edit resource section of built executables (use pefile module already imported into dist.py)
- [ ] Linux: Supply a [Desktop Entry](https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) for each built binary
- [x] macOS: Copy icons into each app bundle

**Comment by Moguri** [comment]
Putting some thought into the interface for this, I suggest an `icons` dict for `build_apps`. The keys are app names (as many other dicts for `build_apps`) with `*` being a fallback for any app. The values are lists of icons which should be various sizes to use. Each platform can determine if wants to use multiple icon sizes or pick the most appropriate size (e.g., largest) for the platform. The other option is to provide a single file and to automatically created scaled-down versions of the icon. I think we should also only support png and svg and we can convert to other formats as necessary (e.g., ico or icns).

**Comment by rdb** [comment]
I wrote an icon class for pdeploy (see `class Icon` in `DeploymentTools.py`) that can handle any format that PNMImage supports.  The code to add an icon to an exe is already in `pefile.py`.  So we just need to wire this up in deploy-ng itself.

**Comment by wezu** [comment]
It would be cool if one could provide the icons at different resolutions, I just don't trust nearest neighbor downsampling to make a good 32x32 icon(or smaller!) from a 256x256 image. Anything under 64x64 is pixel art territory and I'd prefer to have control over how it looks - as in provide my own, separate files for each size and only fall back to scaling a bigger image if there is no fitting size.

**Comment by Moguri** [comment]
Maybe the dict value could be either a filepath or a list of filepaths? That gives people control if they want it, but it keeps things simple for those that may not care.

@rdb Also, good to know about that `Icon` class. I feel like you've told be about it a few times, but I keep forgetting about it.

**Comment by rdb** [comment]
Windows and macOS are implemented and will be in 1.10.4.  Let's open a new issue for the desktop entry support; I would like that it receives a bit more testing in various environments first.

**Comment by rdb** [comment]
@wezu In 1.10.4 (which will be released tomorrow) it is possible to specify a list of icons at different resolutions.  An example demonstrating the syntax follows:
```
    'gui_apps': {'shark_simulator': 'src/main.py'},
    'icons': {
        'shark_simulator': ['hat-shark16.png', 'hat-shark32.png', 'hat-shark48.png', 'hat-shark128.png'],
    },
```
Recommended resolutions to provide for an icon are 16, 32, 48, 128, and 256 pixels, though it doesn't hurt to also provide 24, 64, 512 and 1024 if you have them.


I've opened panda3d/panda3d-docs#11 to track the documentation task.
